### PR TITLE
Support for ISO 14230-2 KWP

### LINKED ISSUE #5: KWP 2000 fast init
Currently only the slow 5 baud OBD9141-2 initialisation procedure is supported. Someone requested KWP2000 fast initialisation, from the specification I attempted to make a blind implementation. This implementation can be found in the [kwp2000_fast_init](../tree/kwp2000_fast_init) branch, it is untested. The added method is `init_kwp_fast`.

Placing this issue here for increased visibility, if anyone can test it and report whether it works or doesn't work that'd be much appreciated. If it doesn't work: pull requests are welcome, as are logic analyzer recordings of another device performing a KWP2000 fast init and recording of this library performing the fast init on the same car.

### LINKED ISSUE #11: Questions and ISO 14230-2 KWP development
First I have to tell your library makes it easy.   So now the questions.  I don't have one of the ICs readily available (they are order be about a week to get) but I found another circuit that is floating around out there on the web, see attached:
obdii_avr

In looking through the lib I know that it can't really be used directly to drive the circuit.  But if I wanted to use this circuit instead of the chip I would have to change several functions.  Any suggestions?

BTW I am using the inti_kwp_fast fork.

Thanks
Mike

### PR DESCRIPTION
**Pull request by iwanders**
This adds the ISO 14230-2 KWP support to the `master` branch. Confirmed today that the 9141-2 functionality on this branch is unaffected using the Display example. This closes #11, closes #5

### CODE REVIEW

File: examples/T3_ILI9341_display/T3_ILI9341_display.ino [hunk 1]
```diff
@@ -70,7 +70,7 @@ typedef struct{
 } obd_unit;
 
 typedef struct{
-  uint32_t old_value;
+  int32_t old_value;
```
Comment (iwanders) [inline examples/T3_ILI9341_display/T3_ILI9341_display.ino:73]: One small unrelated fix to ensure that clearing the previous number doesn't break when a negative number was displayed, such as low temperatures.

File: src/OBD9141.h [hunk 2]
```diff
@@ -57,7 +57,7 @@
 // The ECU might not push all bytes on the bus immediately, but wait several ms
 // between the bytes, this is the time allowed per byte for the answer
 
-#define OBD9141_WAIT_FOR_REQUEST_ANSWER_TIMEOUT (30 + 10)
+#define OBD9141_WAIT_FOR_REQUEST_ANSWER_TIMEOUT (30 + 20)
```
Comment (iwanders) [inline src/OBD9141.h:60]: Slight increase in this timeout, KWP may require some different timing. In normal operation the code never reaches this timeout so it doesn't negatively affect any successful operation because we return after reading all the expected bytes.

**Comment by iwanders** [comment]
@mjs513, this functionality was in large part made possible by your efforts in testing my code. I enjoyed the quick back and forths we did on #11. Thanks for the collaboration and your help in improving this library!

If you have any last comments before this is merged with `master`, let me know. If you would like an explicit callout in the readme to [mjs513](https://github.com/mjs513), feel free to say so and I'll happily add it.

**Comment by mjs513** [comment]
@iwanders.  Sorry for the delay in getting back to you.  I agree I had a lot fun working on this with you testing the code changes.

I really don't have anything else to add before you do the merge.  An no I do not need any explicit callout in the readme. But thanks for offering.

Mike
